cmd/k8s-operator/svc.go

```diff
@@ -187,6 +188,14 @@ func (a *ServiceReconciler) maybeProvision(ctx context.Context, logger *zap.Suga
 		sts.TailnetTargetIP = ip
 		a.managedEgressProxies.Add(svc.UID)
 		gaugeEgressProxies.Set(int64(a.managedEgressProxies.Len()))
+	} else if fqdn := svc.Annotations[AnnotationTailnetTargetFQDN]; fqdn != "" {
+		fqdn := svc.Annotations[AnnotationTailnetTargetFQDN]
+		if !strings.HasSuffix(fqdn, ".") {
+			fqdn = fqdn + "."
+		}
```

Comment:
Could/should we check (here or elsewhere) that provided name is an FQDN rather than a short host name?

Reply:
Added a check to the operator https://github.com/tailscale/tailscale/pull/10360/commits/bb5185d3d5e336035e2ca3dc2140aa3a17628587#diff-eafeab683fd7406e5cb805b7a7c376e35404c1be947f68741c95038ceee08d9aR152-R157 so it does not proceed if the Service annotation is invalid.